components/clp-package-utils/clp_package_utils/scripts/compress_from_s3.py

```diff
@@ -0,0 +1,299 @@
+    args_parser: argparse.ArgumentParser,
+) -> None:
+    """
+    Validates s3-object subcommand arguments.
+    """
+    if len(parsed_args.inputs) == 0 and parsed_args.inputs_from is None:
+        args_parser.error("No URLs specified.")
+
+    # Validate URLs were specified using only one method
+    if len(parsed_args.inputs) > 0 and parsed_args.inputs_from is not None:
+        args_parser.error("URLs cannot be specified on the command line AND through a file.")
+
+
+def _validate_s3_key_prefix_args(
+    parsed_args: argparse.Namespace,
+    args_parser: argparse.ArgumentParser,
+) -> None:
+    """
+    Validates s3-key-prefix subcommand arguments.
+    """
+
+    if parsed_args.inputs_from is None:
+        if len(parsed_args.inputs) == 0:
+            args_parser.error("No URL specified.")
+        if len(parsed_args.inputs) != 1:
+            args_parser.error(
+                f"s3-key-prefix accepts exactly one URL, got {len(parsed_args.inputs)}."
+            )
+
+    if len(parsed_args.inputs) > 0 and parsed_args.inputs_from is not None:
+        args_parser.error("URL cannot be specified on the command line AND through a file.")
+
+
+def main(argv):
+    clp_home = get_clp_home()
+    default_config_file_path = clp_home / CLP_DEFAULT_CONFIG_FILE_RELATIVE_PATH
+
+    args_parser = argparse.ArgumentParser(description="Compresses logs from S3")
+
+    args_parser.add_argument(
+        "--config",
+        "-c",
+        default=str(default_config_file_path),
+        help="CLP package configuration file.",
+    )
+    args_parser.add_argument(
+        "--verbose",
+        "-v",
+        action="store_true",
+        help="Enable debug logging.",
+    )
+    args_parser.add_argument(
+        "--dataset",
+        type=str,
+        default=None,
+        help="The dataset that the archives belong to.",
+    )
+    args_parser.add_argument(
+        "--timestamp-key",
+        help="The path (e.g. x.y) for the field containing the log event's timestamp.",
+    )
+    args_parser.add_argument(
+        "-t", "--tags", help="A comma-separated list of tags to apply to the compressed archives."
+    )
+    args_parser.add_argument(
+        "--no-progress-reporting", action="store_true", help="Disables progress reporting."
+    )
+
+    subparsers = args_parser.add_subparsers(dest="subcommand", required=True)
+
+    object_parser = subparsers.add_parser("s3-object", help="Compress specific S3 objects")
+    object_parser.add_argument("inputs", metavar="URL", nargs="*", help="S3 object URLs")
+    object_parser.add_argument(
+        "--inputs-from", type=str, help="File containing S3 object URLs (one per line)"
+    )
+
+    prefix_parser = subparsers.add_parser(
+        "s3-key-prefix", help="Compress all objects under S3 key prefix"
+    )
+    prefix_parser.add_argument("inputs", metavar="URL", nargs="*", help="S3 prefix URL")
+    prefix_parser.add_argument("--inputs-from", type=str, help="File containing S3 prefix URL")
```

Comment:
```suggestion
    object_parser = subparsers.add_parser(
        "s3-object", help="Compress specific S3 objects identified by their full URLs."
    )
    object_parser.add_argument("inputs", metavar="URL", nargs="*", help="S3 object URLs.")
    object_parser.add_argument(
        "--inputs-from", type=str, help="A file containing all S3 object URLs to compress."
    )

    prefix_parser = subparsers.add_parser(
        "s3-key-prefix", help="Compress all S3 objects under the key prefix."
    )
    prefix_parser.add_argument("inputs", metavar="URL", nargs="*", help="S3 prefix URL.")
    prefix_parser.add_argument(
        "--inputs-from", type=str, help="A file containing S3 key prefix to compress."
    )
```
- Polished the help notes a bit.
- Don't forget the period, lol